thirdeye-core/src/main/java/ai/startree/thirdeye/alert/AlertDetectionIntervalCalculator.java

```diff
@@ -96,10 +96,13 @@ private static Interval getCorrectedInterval(final long alertId, final long task
     DateTime correctedEnd = taskEnd;
     // apply delay correction
     final Period delay = getDelay(metadata);
-    // if alert has already run: start = lastTimestamp = correctedEnd from last run -> delay was applied
-    // if alert has never run: delay is not important if end-delay > start.
-    // only apply delay on start if end-delay > start is not true
-    correctedEnd = correctedEnd.minus(delay);
+    final DateTime dataWatermark = new DateTime(chronology).minus(delay);
```

Comment:
If I understand correctly, this is using the current time when this line is executed.

I think this is fine. We assume that physically, the data within the delay window isn't complete, so there is an inherent dependency on the time of execution and the data that's operated on. Since this interval correction happens outside/before the detection pipeline execution, the pipeline itself is still repeatable `:copium:`.

Reply:
yes the pipeline execution is still repeatable

Reply:
let me check if there is something clearer like `DateTime.now()`